Modules/Test/classes/class.ilObjTest.php

```diff
@@ -5254,6 +5254,7 @@ public function fromXML(ilQTIAssessment $assessment)
         }
 
         $this->saveToDb();
+        $result_summary_settings = $result_summary_settings->withShowPassDetails($result_details_settings->getShowPassDetails());
```

Comment:
By default, the value of `results_presentation` in the `tst_tests` table is set to `3` if it is not specified. This results in `showPassDetails()` always being true.

Subsequently, `result_presentation` will be set to the corresponding import value (in my case, `0`). However, due to this line 
https://github.com/ILIAS-eLearning/ILIAS/blob/a54812f6c900046a80f48f2675620d54ce182b6f/Modules/Test/classes/ScoreReporting/class.ilObjTestScoreSettingsDatabaseRepository.php#L126-L128
, it will always be set to `1`. The reason for this is that PassDetails is true because it inherits the still-existing default value of 3 from the database.

@nhaagen, you seem to be the author of this feature. Please take a look at this and see if there is a better solution.

Best,
@fhelfer

Reply:
This seems to also be an issue for newer releases